tracing/src/span.rs

```diff
@@ -679,7 +679,29 @@ impl Span {
     ///     }
     /// }
     /// ```
+    ///
+    /// Beware though!
+    /// The fields associated with a span are part of its [`Metadata`].
+    /// The [`Metadata`] describing a particular span is constructed statically when the span is
+    /// created and cannot be extended later to add new fields.
+    /// Hence you cannot record a value for a field that was not specified at the moment
+    /// of span creation:
+    /// ```
+    /// use tracing::{trace_span, field};
+    ///
+    /// // Create a span with two fields: `greeting`, with the value "hello world", and
+    /// // `parting`, without a value.
+    /// let span = trace_span!("my_span", greeting = "hello world", parting = field::Empty);
+    ///
+    /// // ...
+    ///
+    /// // Now, you try to record a value for a new field, `new_field`, which was not
+    /// // declared as `Empty` or populated when you created `span`.
+    /// // You won't get any error, but the assignment will have no effect!
+    /// span.record("new_field", &"interesting_value_you_really_need");
+    /// ```
```

Comment:
In general, with this kind of note, I like to conclude by re-stating what users _should_ do instead to avoid problems, even though this is already explained in the docs before the note. So, I think it might be good to end with something like "Instead, all fields that may be recorded should be declared up front, using `field::Empty` when a value is not known". 

What do you think?

Reply:
That sounds good - we could put it on top a new line in the example where we register a value for `parting`. It partially repeats the example above, but I think it would help driving the point home. What do you think?

Reply:
I think a little repetition is just fine.

Reply:
Done :+1: